src/nvOCDR.cpp

```diff
@@ -311,9 +322,26 @@ nvOCDR::nvOCDR(nvOCDRParam param):
                                                                  mOCRNetMaxBatch,
                                                                  upsidedown, isNHWC,
                                                                  param.rotation_threshold)));
-    // Set the OCRNet's TRT input buffer as rect's output
-    mRect->setOutputBuffer(mOCRNet->mTRTInputBufferIndex);
+
     mRect->initBuffer(mBuffMgr);
+    // Set rect's output as the OCRNet's TRT input buffer
+    if (mOCRNetInputShape.d[1] == 1)
+    {
+        // channel = 1 means OCRnet use gray image as input
+        int grayRectOutputDevIdx = mRect->getGrayOutputDevBufferIdx();
+        mOCRNet->setInputDeviceBuffer(mBuffMgr.mDeviceBuffer[grayRectOutputDevIdx], grayRectOutputDevIdx);
+    }
+    else if (mOCRNetInputShape.d[1] == 3)
+    {
+        // channel = 3 means OCRnet use rgb image as input
+        int rgbRectOutputDevIdx = mRect->getRGBOutputDevBufferIdx();
+        mOCRNet->setInputDeviceBuffer(mBuffMgr.mDeviceBuffer[rgbRectOutputDevIdx], rgbRectOutputDevIdx);
```

Comment:
I suppose The RGB buffer should be in float precision instead of uchar format?

Reply:
And why do you need to initialize the buffer in `RectEngine` instead of `OCRNetEngine`?

Reply:
RGB buffer should be float, I'll change it.

OCR_resenet and OCR_vit are using gray input, but CLIP4STR is using RGB as input, RectEngine will product gray and RGB data buffer, so I directly set the OCRNetEngine 's input from RectEngine according the input channel